add docs for response action apis (#2232)
elastic/security-docs: docs/management/api/suspend-process-api.asciidoc

[[suspend-process-api]]
=== Suspend a process

> Suspend a process on a host running {endpoint-cloud-sec}.

Reviewer: Is there a way to resume a suspended process? And why would a user suspend a process instead of terminate it, or vice versa?

Asking partly on my own unfamiliarity with the features, but also I think customers might have similar questions, especially about how to resume a process.

Author: > Is there a way to resume a suspended process?

Currently, not through us. I believe you'd need to do it on the host.

> And why would a user suspend a process instead of terminate it, or vice versa?

My knowledge is limited on this but my guess is that for certain threats you may need to suspend a number its watch processes prior to killing the desired processes so it can't defend against termination. I'm probably missing things though so it maybe best to have someone else to chime in on this.

> Asking partly on my own unfamiliarity with the features, but also I think customers might have similar questions, especially about how to resume a process.

I believe our target audience that is expected to have access to response actions will be pretty familiar with suspend/kill as they are a common part of their toolbox for responding to threats. At the minimum, I expect them to be more familiar with responding to threats than me 😅.

Reviewer: Thanks for the extra context! I agree, the target audience will probably have more familiarity than you or me!

> You must have the `superuser` role to perform this action and at least an Enterprise license.

Reviewer 2: ```suggestion
You must have the `superuser` role and at least an Enterprise license to perform this action.
```

Reviewer 2: If you accept this suggestion, it also applies elsewhere in the PR.

==== Request URL

`POST <kibana host>:<port>/api/endpoint/action/suspend_process`

==== Request body

A JSON object with these fields:

[width="100%",options="header"]
|==============================================
|Name |Type |Description |Required

> |`endpoint_ids` |Array (String) |The IDs of each endpoint you want to issue this action to. |Yes

Reviewer 2: ```suggestion
|`endpoint_ids` |Array (String) |The ID(s) of endpoint(s) you want to issue this action to. |Yes
```
Assuming it could be just one endpoint, this applies elsewhere in the PR as well.

|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here. The action will be logged in any cases associated with the specified alerts. |No
|`case_ids` |Array (String) |Logs the action taken on specified cases. |No
|`comment` |String |Attach a comment to this action's log. The comment text will appear in associated cases. |No

> |`parameters.pid` |Number |The pid of the process to suspend. |Yes* (Either `parameters.pid` or `parameters.entity_id` must be provided, but not both)

Reviewer 2: ```suggestion
|`parameters.pid` |Number |The PID of the process to suspend. |Yes* (Either `parameters.pid` or `parameters.entity_id` must be provided, but not both)
```

> |`parameters.entity_id` |String |The entity id of the process to suspend. |Yes* (Either `parameters.pid` or `parameters.entity_id` must be provided, but not both)

Reviewer 2: ```suggestion
|`parameters.entity_id` |String |The entity ID of the process to suspend. |Yes* (Either `parameters.pid` or `parameters.entity_id` must be provided, but not both)
```

|==============================================

===== Example requests

> Suspends the process with entity_id `abc123` on a host with an `endpoint_id` value of `ed518850-681a-4d60-bb98-e22640cae2a8` and comment of `suspend the process`:

Reviewer 2: ```suggestion
Suspends the process with entity_id `abc123` on a host with an `endpoint_id` value of `ed518850-681a-4d60-bb98-e22640cae2a8` and comments `suspend the process`:
```

[source,sh]
--------------------------------------------------
POST /api/endpoint/action/suspend_process
{
  "endpoint_ids": ["ed518850-681a-4d60-bb98-e22640cae2a8"],
  "parameters": {
    "entity_id": "abc123"
  },
  "comment": "suspend the process"
}
--------------------------------------------------
// KIBANA

===== Response code

`200`::
   Indicates a successful call.

`403`::
	Indicates insufficient user role (must be `superuser`), or unsupported license level (minimum Enterprise license required).

`500`::

> 	General error. A response message will indicate the failure.

Reviewer 2: ```suggestion
	General error. A response message will provide additional details.
```

Reviewer 2: if accepted, this suggestion applies elsewhere in the PR
